js/src/wasm/moz.build

```diff
@@ -48,6 +48,9 @@ UNIFIED_SOURCES += [
     "WasmValType.cpp",
     "WasmValue.cpp",
 ]
+SOURCES += [
+    "WasmCode-platform.cpp",
```

Comment:
In the previous commit where I spun `WasmCode-platform.cpp` into a new file, I forgot to add it to the build system, which broke compilation outside of the `mongo` code base.

I'm adding it here instead of directly to `UNIFIED_SOURCES`, because adding it to `UNIFIED_SOURCES` would require us to update the `src/third_party/mozjs/platform/<ARCH>/<PLATFORM>/build/Unified_cpp_js_src_wasm*.cpp` files for every platform.

@mdashti Unless we decide to bite the bullet and add this to the `UNIFIED_SOURCES` list, we'll have to also add this file to `src/third_party/mozjs/SConstruct`.

Reply:
@jseyster IMO, it's OK to keep it outside `UNIFIED_SOURCES`, as it requires a lot of effort to regenerate the configurations for all platform.